client/src/modules/Admin/Components/Stats.tsx

```diff
@@ -73,56 +68,33 @@ export default class Stats extends Component<Props, State> {
       })
   }
 
-  howManyReviewsEachClass() {
-    axios
-      .post(`/api/howManyReviewsEachClass`, {
-        token: this.props.token,
-      })
-      .then((res) => {
-        let data = res.data.result
-        data.sort((rev1: any, rev2: any) => (rev1.total > rev2.total ? -1 : 1))
-        this.setState({ howManyReviewsEachClass: data })
-      })
-      .catch((err) => {
-        console.log('error retrieving reviews for each class ', err)
-      })
-  }
+  numReviewsPerClass(reviews: Review[]): {courseName?: string, reviewCount?: number | undefined}[] {
+    const reviewsPerCourse: Map<string | undefined, number | undefined> = new Map()
 
-  getReviewsPerClassCSV() {
-    let strRet = 'class,total\n'
-    this.state.howManyReviewsEachClass.forEach((obj) => {
-      strRet += obj._id + ',' + obj.total + '\n'
+    reviews.forEach(
+      review => {
+        if (reviewsPerCourse.has(review.class!)) {
+          reviewsPerCourse.set(review.class!, reviewsPerCourse.get(review.class!)! + 1)
+        } else {
+          reviewsPerCourse.set(review.class!, 1)
+        }
+      }
+    )
+    
+    const result: {courseName: string | undefined, reviewCount: number | undefined}[] = [];
+    reviewsPerCourse.forEach((count, courseName) => {
+      result.push({courseName, reviewCount: count})
     })
-    return strRet
+    return result
   }
 
-  getHowManyEachClass() {
-    axios
-      .post(`/api/howManyEachClass`, {
-        token: this.props.token,
-      })
-      .then((res) => {
-        let data = res.data.result
-        data.sort((rev1: any, rev2: any) => (rev1.total > rev2.total ? -1 : 1))
-        this.setState({ howManyEachClass: data })
-      })
-      .catch((err) => {
-        console.log('error retrieving how many each class ', err)
-      })
-  }
-
-  totalReviews() {
-    axios
-      .post(`/api/totalReviews`, {
-        token: this.props.token,
-      })
-      .then((res) => {
-        const total = res.data.result
-        this.setState({ totalReviews: total })
-      })
-      .catch((err) => {
-        console.log('error retrieving totalReviews ', err)
-      })
+  getReviewsPerClassCSV() {
```

Comment:
Also I think for the CSV, since className shows the database id of the class, maybe you can make another endpoint after calling /fetchAllReviews that will get the class name from the classes database using the class id stored in each review and then map that to the review count instead.